src/test/java/com/google/api/generator/engine/writer/JavaWriterVisitorTest.java

```diff
@@ -89,6 +89,29 @@ public void writeAnnotation_withDescription() {
   }
 
   /** =============================== EXPRESSIONS =============================== */
+  @Test
+  public void writeStringObjectValue() {
+    StringObjectValue s = StringObjectValue.builder().setValue("\"test\"").build();
+    assertThat(s.value()).isEqualTo("\"test\"");
+    assertThat(s.type()).isEqualTo(TypeNode.STRING);
+  }
+
+  @Test
+  public void writeStringObjectValue_assignmentExpr() {
+    IdentifierNode identifier = IdentifierNode.builder().setName("x").build();
+    Variable variable = Variable.builder().setIdentifier(identifier).setType(TypeNode.STRING).build();
+    VariableExpr variableExpr =
+        VariableExpr.builder().setVariable(variable).setIsDecl(true).build();
+
+    Value value = StringObjectValue.withValue("\"test\"");
```

Comment:
In the expected behavior below, the caller should be doing `withValue("test")` IMO.